src/refract/SerializeCompactVisitor.cc

```diff
@@ -0,0 +1,87 @@
+//
+//  refract/SerializeCompactVisitor.cc
+//  librefract
+//
+//  Created by Jiri Kratochvil on 21/05/15.
+//  Copyright (c) 2015 Apiary Inc. All rights reserved.
+//
+#include "Element.h"
+
+namespace refract
+{
+
+    void SerializeCompactVisitor::visit(const IElement& e)
+    {
+        throw NotImplemented("NI: IElement Compact Serialization");
+    }
+
+    void SerializeCompactVisitor::visit(const NullElement& e)
+    {
+        value_ = sos::Null();
+    }
+
+    void SerializeCompactVisitor::visit(const StringElement& e)
+    {
+        value_ = sos::String(e.value);
+    }
+
+    void SerializeCompactVisitor::visit(const NumberElement& e)
+    {
+        value_ = sos::Number(e.value);
+    }
+
+    void SerializeCompactVisitor::visit(const BooleanElement& e)
+    {
+        value_ = sos::Boolean(e.value);
+    }
+
+    void SerializeCompactVisitor::visit(const ArrayElement& e)
+    {
+        sos::Array array;
+        typedef ArrayElement::ValueType::const_iterator iterator;
+        for (iterator it = e.value.begin(); it != e.value.end(); ++it) {
```

Comment:
Difficult to read through. Please use empty lines before and after `for` and `if` blocks.